### PR TITLE
Simple lock: Add endpoints that breaks farm

### PR DESCRIPTION
**Pull request by CostinCarabas**
exit_farm_locked_token_to_tokens - will break farm into lp and lp into initial tokens

### CODE REVIEW

File: locked-asset/proxy_dex/src/proxy_farm.rs [hunk 1]
```diff
@@ -234,6 +234,80 @@ pub trait ProxyFarmModule:
...
+        output_payments.push(exit_result.reward_tokens.clone());
+
+        self.send_multiple_tokens_if_not_zero(&caller, &output_payments);
+
+        self.emit_exit_farm_proxy_event(
+            &original_caller,
+            &farm_address,
+            payment,
+            wrapped_farm_attributes_for_exit,
+            exit_result.reward_tokens.clone(),
```
Comment (dorin-iancu) [inline locked-asset/proxy_dex/src/proxy_farm.rs:305]: I think you don't need this `clone`.

File: locked-asset/proxy_dex/src/proxy_farm.rs [hunk 2]
```diff
@@ -324,6 +398,9 @@ pub trait ProxyFarmModule:
         self.send_payment_non_zero(&caller, &new_wrapped_token);
         self.send_payment_non_zero(&caller, &claim_result.rewards);
 
+        // Burn farm token
+        wrapped_farm_token_mapper.nft_burn(payment.token_nonce, &payment.amount);
```
Comment (dorin-iancu) [inline locked-asset/proxy_dex/src/proxy_farm.rs:402]: It seems like you're burning the farm token twice? (other burn on line 293)

File: locked-asset/proxy_dex/src/proxy_pair.rs [hunk 3]
```diff
@@ -143,12 +143,32 @@ pub trait ProxyPairModule:
...
+        output_payments.into()
+    }
+
+    fn remove_liquidity_proxy_common(
+        &self,
+        input_payment: EsdtTokenPayment,
+        pair_address: ManagedAddress,
+        first_token_amount_min: BigUint,
+        second_token_amount_min: BigUint,
+    ) -> ManagedVec<EsdtTokenPayment> {
```
Comment (dorin-iancu) [inline locked-asset/proxy_dex/src/proxy_pair.rs:165]: Struct with named fields, same in endpoint.

File: locked-asset/simple-lock/src/proxy_farm.rs [hunk 4]
```diff
@@ -248,6 +250,93 @@ pub trait ProxyFarmModule:
...
+    /// Output Payments:
+    /// - original tokens
+    /// - farm reward tokens
+    #[payable("*")]
+    #[endpoint(destroyFarmLockedTokens)]
+    fn destroy_farm_locked_tokens(
+        &self,
+        first_token_min_amount_out: BigUint,
+        second_token_min_amount_out: BigUint,
+    ) -> ExitFarmThroughProxyResultType<Self::Api> {
```
Comment (dorin-iancu) [inline locked-asset/simple-lock/src/proxy_farm.rs:266]: struct

**Review by dorin-iancu — DISMISSED** [review]

**Review by psorinionut — APPROVED** [review]

**Review by psorinionut — APPROVED** [review]

**Pull request merged**